client/src/components/WebAuthn/Verify.js

```diff
@@ -2,87 +2,218 @@
-                : null,
-            },
-          })),
-        });
-      })
-      .catch(error => {
-        console.log('nay', error, error.message);
-      });
+    this.startAuth = this.startAuth.bind(this);
+    this.handleRetry = this.handleRetry.bind(this);
   }
 
-  handleStartAuth(event) {
+  handleRetry(event) {
     event.preventDefault();
-    this.initAuth();
+    this.setState({
+      failure: false,
+    });
   }
 
   /**
-   * Render a description for this input
+   * Begins the Webauthn authentication process - presents the browser prompt to the user
+   * This happens by delegating to {@see performVerification} which returns a promise that will
+   * be completed after the process is complete.
+   */
+  startAuth() {
+    const { publicKey, onCompleteVerification } = this.props;
+
+    performVerification(publicKey)
+      .then(verificationData => {
+        this.setState(
+          { success: true },
+          () => setTimeout(
+            () => { onCompleteVerification(verificationData); },
+            1000
+          )
+        );
+      })
+      .catch(() => this.setState({ failure: true }));
+  }
+
+  /**
+   * Render instructions for registering with this method
    *
    * @return {HTMLElement}
    */
   renderDescription() {
     const { ss: { i18n } } = window;
+    const { method: { supportLink } } = this.props;
+
+    const registerKeyT = i18n._t(
+      'MFAWebAuthnVerify.VERIFY',
+      fallbacks['MFAWebAuthnVerify.VERIFY']
+    );
+
+    // As this part of the message requires formatting, we render it using dangerouslySetInnerHTML
+    const instructions = i18n.inject(
+      i18n._t('MFAWebAuthnVerify.INSTRUCTION', fallbacks['MFAWebAuthnVerify.INSTRUCTION']),
+      { button: `<strong>${registerKeyT}</strong>` }
+    );
 
     return (
-      <p>
-        {i18n._t(
-          'MFAWebAuthnVerify.DESCRIPTION',
-          'Use your security key'
-        )}
-      </p>
+      <div className="mfa-verification-container__description">
+        <p>
+          {i18n._t('MFAWebAuthnVerify.DESCRIPTION', fallbacks['MFAWebAuthnVerify.DESCRIPTION'])}
+
+          {supportLink &&
+            <a
+              href={supportLink}
+              target="_blank"
+              rel="noopener noreferrer"
+            >
+              {i18n._t('MFAWebAuthnVerify.HELP', fallbacks['MFAWebAuthnVerify.HELP'])}
+            </a>
+          }
+        </p>
+
+        {/* eslint-disable-next-line react/no-danger */}
+        <p dangerouslySetInnerHTML={{ __html: instructions }} />
+      </div>
     );
   }
 
+  /**
+   * Render the status of the registration, with relevant iconography
+   *
+   * @returns {HTMLElement}
+   */
+  renderStatus() {
+    const { ss: { i18n } } = window;
+    const { errors } = this.props;
+    const { failure, success } = this.state;
 
-  render() {
+    if (errors.length) {
+      return (
+        <div className="mfa-verification-container__status status-message--error">
```

Comment:
Just noting that it doesn't look like child authenticator Verify components get passed the `errors` prop correctly, as it does in Register components. A problem for another time though, the existing failure message says to try inserting your key again, which doesn't help you if you've timed out from sudo mode etc, but better than nothing.